staging/src/k8s.io/client-go/tools/clientcmd/loader.go

```diff
@@ -129,6 +129,20 @@ type ClientConfigLoadingRules struct {
 	// WarnIfAllMissing indicates whether the configuration files pointed by KUBECONFIG environment variable are present or not.
 	// In case of missing files, it warns the user about the missing files.
 	WarnIfAllMissing bool
+
+	// Warner is the warning log callback to use in case of missing files.
+	Warner WarningHandler
```

Comment:
i think WarningHandler is easier to read than Warner

Reply:
Do you suggest to use WarningHandler as instance and Warner as type?

Reply:
what about using: WarningHandler WarningHandlerFunc